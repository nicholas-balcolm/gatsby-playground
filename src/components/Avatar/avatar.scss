$primary: green;
$secondary: lighten($primary, 20%);
$black: #000;
$white: #fff;

.avatar {
    margin: 22px;
    text-align: center;
    position: relative;
    display: inline-block;
    
    &:hover img {
        transition: all .2s ease-out;
        border: solid 5px $primary;
        border-radius: 100px;
        box-shadow: 0px 5px 10px rgba($black,.5);
    }

    span {
        display: block;
        text-align: center;
        font-size: 30px;
        font-weight: bold;
    }

    img {
        box-sizing: border-box;
        border-radius: 10px;
        border: solid 2px $secondary;
        transition: all .2s ease-in;
    }

    .user-name {
        position: absolute;
        top: 90%;
        left: 50%;
        transform: translateY(-50%);
        transform: translateX(-50%);
        color: $white;
        font-size: 18px;
        text-shadow: 0px 0px 10px $black;
    }
}