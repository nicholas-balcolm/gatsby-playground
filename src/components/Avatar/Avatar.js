import React, {Component} from 'react';
import './avatar.scss';

class Avatar extends Component {
    constructor(props){
        super(props);
        this.state = {
            'id': props.id,
            'alt': props.alt,
            'user': props.user
        };
    }

    render() {
        return (
            <div className="avatar">
                <span className="user-name">{this.props.user}</span>
                <a href={'https://github.com/' + this.state.user}>
                    <img
                        src={this.getPhotoUrl()}
                        alt={this.state.alt}
                        className="commentPhoto widget" /><br />
                </a>
            </div>
        );
    }

    getPhotoUrl() {
        return 'https://avatars.githubusercontent.com/' + this.props.user + '?s=200';
    }
}

export default Avatar;