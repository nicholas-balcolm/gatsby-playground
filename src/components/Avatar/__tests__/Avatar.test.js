jest.unmock('../Avatar');
import React from 'react';
import Avatar from '../Avatar';
import renderer from 'react-test-renderer';

describe('the avatar component', () => {
    const props = {
        id: 2,
        user: 'dfederspiel',
        alt: 'photo of '
    };

    it('renders correctly', () => {
        const tree = renderer.create(
            <Avatar
                {...props}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('given a user name', () => {
        it('should display the users avatar', () => {

            const tree = renderer.create(
                <Avatar
                    {...props}
                />
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});