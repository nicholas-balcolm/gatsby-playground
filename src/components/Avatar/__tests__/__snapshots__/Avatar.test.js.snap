// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the avatar component given a user name should display the users avatar 1`] = `
<div
  className="avatar"
>
  <span
    className="user-name"
  >
    dfederspiel
  </span>
  <a
    href="https://github.com/dfederspiel"
  >
    <img
      alt="photo of "
      className="commentPhoto widget"
      height={200}
      src="https://avatars.githubusercontent.com/dfederspiel?s=200"
      width={200}
    />
    <br />
  </a>
</div>
`;

exports[`the avatar component renders correctly 1`] = `
<div
  className="avatar"
>
  <span
    className="user-name"
  >
    dfederspiel
  </span>
  <a
    href="https://github.com/dfederspiel"
  >
    <img
      alt="photo of "
      className="commentPhoto widget"
      height={200}
      src="https://avatars.githubusercontent.com/dfederspiel?s=200"
      width={200}
    />
    <br />
  </a>
</div>
`;
